### PR TITLE
603 - Update OdsChip OdsChoiceChipsFlowRow and OdsFilterChip APIs

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: lib/src/main/java/com/orange/ods/extension/ContextExt.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+package com.orange.ods.extension
+
+import android.content.Context
+
+/**
+ * @return the dp corresponding to the given [pixelSize]
+ */
+fun Context.getDpFromPx(pixelSize: Int) = pixelSize / this.resources.displayMetrics.density
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/extension/ContextExt.kt:18]: This don't take into account the resolution: 
```
/**
 * Create a [Dp] using an [Int]:
 *     val left = 10
 *     val x = left.dp
 *     // -- or --
 *     val y = 10.dp
 */
```
I tested but the image is too big

[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/extension/ContextExt.kt:18]: My bad, it's ok like this:
```kotlin
@OptIn(ExperimentalMaterialApi::class)
fun Modifier.odsChipAvatar(enabled: Boolean): Modifier = composed {
    this.size(dimensionResource(R.dimen.icon_size))
        .clip(CircleShape)
        .alpha(if (enabled) 1f else ChipDefaults.LeadingIconOpacity)
}
```

File: lib/src/main/java/com/orange/ods/compose/component/chip/OdsChip.kt [hunk 2]
```diff
@@ -162,10 +174,10 @@ private fun PreviewOdsChip(@PreviewParameter(OdsChipPreviewParameterProvider::cl
     var selected by remember { mutableStateOf(false) }
     OdsChip(
         text = "Text",
+        outlined = outlined,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/chip/OdsChip.kt:177]: I had forgotten the `previewThemeConfigurationClass`, so I remove the parameters from previews

File: lib/src/main/java/com/orange/ods/compose/component/chip/OdsChoiceChipsFlowRow.kt [hunk 3]
```diff
@@ -22,37 +22,35 @@ import androidx.compose.ui.res.dimensionResource
...
- * @param outlinedChips If true, a border will be drawn around [FlowRow] chips. Otherwise chips will be filled.
  * @param content The content of the choice chips [FlowRow].
  */
 @Composable
 @OdsComposable
 fun <T> OdsChoiceChipsFlowRow(
     selectedChip: MutableState<T>,
     modifier: Modifier = Modifier,
-    outlinedChips: Boolean = true,
     content: @Composable OdsChoiceChipsFlowRowScope<T>.() -> Unit
```
[the remark this answers is not in the document]
Comment (florentmaitre) [reply lib/src/main/java/com/orange/ods/compose/component/chip/OdsChoiceChipsFlowRow.kt]: I think we should not allow the user to modify the `modifier` of the choice chip. That's the first time we encounter this case thus I am not sure about the solution but we could replace the `modifier` parameter with `semantics`:

```kotlin
class OdsChoiceChip<T>(
    val text: String,
    val value: T,
    val enabled: Boolean = true,
    val semantics: SemanticsPropertyReceiver.() -> Unit = {}
) {

    @Composable
    fun Content(selected: Boolean, onSelectedStateChange: (selected: Boolean) -> Unit) {
        val chipStateDescription = selectionStateDescription(selected = selected)
        OdsChip(
            text = text,
            modifier = Modifier.semantics {
                stateDescription = chipStateDescription
                semantics()
            },
            selected = selected,
            onClick = { onSelectedStateChange(!selected) },
            enabled = enabled
        )
    }
}
```

And also update the Kdoc.

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**